### PR TITLE
Initial linter setup

### PR DESCRIPTION
**Pull request by hanneshapke**
## Description

This PR adds a pylint setup to the repo. The Github actions are configured to run for every push to Github.
I took over the TFX Addons styles provided by @casassg, but had to make 3 adjustments to comply with the existing code base:
* line length increased from 80 to 125 characters
* added abstract-class-instantiated to exclusion list
* added protected-access to exclusion list  

[Successful passed linter step](https://github.com/digits/model-card-toolkit/actions/runs/3452837655/jobs/5762999118)

## Checklist

Before submitting a pull request, please be sure to do the following:
- [ ] Read the [How to Contribute guide](https://github.com/tensorflow/model-card-toolkit/blob/master/CONTRIBUTING.md)
      if this is your first contribution.
- [ ] Open an issue to discuss this change.
- [ ] Update documentation if applicable.
- [ ] Write new tests if applicable.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Lint
+
+on:
+  push:
+    paths:
+      - '**.py'
+      - '.github/ci/**'
+      - '.github/workflows/lint.yml'
+    branches:
+      - '**'
```
Comment (hanneshapke) [inline .github/workflows/lint.yml:10]: OK to always run it?

File: model_card_toolkit/core.py [hunk 2]
```diff
@@ -404,11 +402,10 @@ def export_format(self,
                          'Call scaffold_assets() to generate model_card.')
 
     # Generate Model Card.
-    jinja_env = jinja2.Environment(
```
Comment (codesue) [inline model_card_toolkit/core.py:407]: Personally, I prefer how this was before. I find the previous way easier to read, easier tell what changed when updated, and it leaves more room for longer statements. Which rule covers this?

File: model_card_toolkit/utils/validation_test.py [hunk 3]
```diff
@@ -16,13 +16,15 @@
...
-from absl.testing import absltest
-from absl.testing import parameterized
+
 import jsonschema
+from absl.testing import absltest, parameterized
+
 from model_card_toolkit.utils import validation
 
 _MODEL_DETAILS_V1_DICT = {
-    "name": "my model",
```
Comment (codesue) [inline model_card_toolkit/utils/validation_test.py:25]: Is this intentional?

**Review by codesue — COMMENTED** [review]
For the three adjustments, how feasible is it to fix existing code and/or add inline exclusions where quick fixes aren't possible? Ideally, we'd want to prevent introducing code that goes against these rules.

File: setup.py [hunk 4]
```diff
@@ -17,15 +17,13 @@
...
 import subprocess
+# TODO(b/188859752): deprecate distutils
+from distutils.command import build
 
-from setuptools import Command
-from setuptools import setup
+from setuptools import Command, setup
 
 # TODO(b/174880612): reduce dependency resolution search space
 REQUIRED_PACKAGES = [
```
Comment (codesue) [inline setup.py:29]: Should we add lint deps to tests_require to enable local linting?

File: .github/workflows/lint.yml (hunk 1 above)
Comment (casassg) [reply .github/workflows/lint.yml:10]: I dont think you need it on push on every branch tbh

**Comment by hanneshapke** [comment]
PR diverged to far from main

**Pull request closed without merging**